runtime/src/dispatcher.rs

```diff
@@ -253,6 +269,11 @@ impl Dispatcher {
                         signed_policy_raw,
                     );
                 }
+                Ok((_ctx, _id, Body::RuntimeAbortRequest {})) => {
+                    info!(self.logger, "Received abort request");
+                    txn_dispatcher.abort_batch();
```

Comment:
This won't work as the txn dispatcher executes in the same thread as where you process the abort request. The `abort_batch` would need to be called from a different thread (e.g., the protocol thread). (It is fine if you handle the request here to notify the requestor that abort is done, it is just that this will currently always wait for the whole batch to complete processing.)

One of the problems with that is that the transaction dispatcher is not reachable from there.